### PR TITLE
[Java] Add recoverWithNull to JSONOptions and pass to Table.readJSON

### LINKED ISSUE #14073: [FEA] [JNI] Add `recovery_mode` option to `Java_ai_rapids_cudf_Table_readJSON`
**Is your feature request related to a problem? Please describe.**
I would like to be able to specify the recovery mode as either `RECOVER_WITH_NULL` or `FAIL` when calling `Table.readJSON` from the Spark plugin.

**Describe the solution you'd like**
Add a new parameter to `Java_ai_rapids_cudf_Table_readJSON` (and the Java binding) and pass it through to the `json_reader_options_builder`. Given that there are only two options currently, I propose adding a boolean `recover_with_null` parameter.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by andygrove**
## Description

This PR exposes the recently added `json_reader_options_builder::recovery_mode` option in the JNI layer.

closes #14073

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 1]
```diff
@@ -42,6 +42,7 @@
 import org.apache.parquet.schema.OriginalType;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
```
Comment (gerashegalov) [inline java/src/test/java/ai/rapids/cudf/TableTest.java:45]: unused import?

Comment (andygrove) [reply java/src/test/java/ai/rapids/cudf/TableTest.java]: yes, thank you

**Review by gerashegalov — APPROVED** [review]
LGTM

File: java/src/main/java/ai/rapids/cudf/JSONOptions.java [hunk 2]
```diff
@@ -44,6 +46,10 @@ public boolean isLines() {
     return lines;
   }
 
+  public boolean isRecoverWithNull() {
```
Comment (razajafri) [inline java/src/main/java/ai/rapids/cudf/JSONOptions.java:49]: javadocs for public method

File: java/src/main/java/ai/rapids/cudf/JSONOptions.java [hunk 3]
```diff
@@ -78,6 +86,11 @@ public Builder withLines(boolean perLine) {
       return this;
     }
 
+    public Builder withRecoverWithNull(boolean recoverWithNull) {
```
Comment (razajafri) [inline java/src/main/java/ai/rapids/cudf/JSONOptions.java:89]: same comment as above

File: java/src/main/java/ai/rapids/cudf/JSONOptions.java (hunk 2 above)
Comment (andygrove) [reply java/src/main/java/ai/rapids/cudf/JSONOptions.java:50]: Thanks. I have added javadoc.

**Review by revans2 — APPROVED** [review]

**Review by razajafri — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]

**Comment by ttnghia** [comment]
Failed style check:
```
diff --git a/java/src/main/native/src/TableJni.cpp b/java/src/main/native/src/TableJni.cpp
index 0c029a128a..b208ef8f38 100644
--- a/java/src/main/native/src/TableJni.cpp
+++ b/java/src/main/native/src/TableJni.cpp
@@ -1346,7 +1346,8 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Table_readAndInferJSON(
                                         static_cast<std::size_t>(buffer_length)};
 
     auto const recovery_mode = recover_with_null ?
-        cudf::io::json_recovery_mode_t::RECOVER_WITH_NULL : cudf::io::json_recovery_mode_t::FAIL;
+                                   cudf::io::json_recovery_mode_t::RECOVER_WITH_NULL :
+                                   cudf::io::json_recovery_mode_t::FAIL;
     cudf::io::json_reader_options_builder opts = cudf::io::json_reader_options::builder(source)
                                                      .dayfirst(static_cast<bool>(day_first))
                                                      .lines(static_cast<bool>(lines))
@@ -1453,8 +1454,9 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Table_readJSON(
                                                       static_cast<std::size_t>(buffer_length)} :
                                 cudf::io::source_info{filename.get()};
 
-    cudf::io::json_recovery_mode_t recovery_mode = recover_with_null ?
-        cudf::io::json_recovery_mode_t::RECOVER_WITH_NULL : cudf::io::json_recovery_mode_t::FAIL;
+    cudf::io::json_recovery_mode_t recovery_mode =
+        recover_with_null ? cudf::io::json_recovery_mode_t::RECOVER_WITH_NULL :
+                            cudf::io::json_recovery_mode_t::FAIL;
     cudf::io::json_reader_options_builder opts = cudf::io::json_reader_options::builder(source)
                                                      .dayfirst(static_cast<bool>(day_first))
                                                      .lines(static_cast<bool>(lines))
```

**Comment by gerashegalov** [comment]
```bash
pre-commit run clang-format --all-files
```

**Pull request merged**